src/System.Buffers/src/System/Buffers/DefaultArrayPoolBucket.cs

```diff
@@ -48,9 +52,21 @@ internal T[] Rent()
                 // Check if all of our buffers have been rented
                 if (_index < _data.Length)
                 {
-                    buffer = _data[_index] ?? new T[_bufferLength];
+                    buffer = _data[_index];
+                    if (buffer == null)
```

Comment:
Nit: based on code coverage, it doesn't look like any unit tests cover the case when this is false